src/gui/KM_InterfaceGamePlay.pas

```diff
@@ -2964,11 +2965,16 @@ procedure TKMGamePlayInterface.KeyUp(Key: Word; Shift: TShiftState);
     or gGame.IsMPGameSpeedUpAllowed
     or MULTIPLAYER_SPEEDUP then
   begin
+
     // Game speed/pause: available in multiplayer mode if the only player left in the game
-    if Key = gResKeys[SC_SPEEDUP_1].Key then gGame.SetGameSpeed(1, False);
-    if Key = gResKeys[SC_SPEEDUP_2].Key then gGame.SetGameSpeed(gGameApp.GameSettings.SpeedMedium, True);
-    if Key = gResKeys[SC_SPEEDUP_3].Key then gGame.SetGameSpeed(gGameApp.GameSettings.SpeedFast, True);
-    if Key = gResKeys[SC_SPEEDUP_4].Key then gGame.SetGameSpeed(gGameApp.GameSettings.SpeedVeryFast, True);
+    if gGame.IsMPGameSpeedUpAllowed then

```

Comment:
Again I'm not sure why IsMPGameSpeedUpAllowed being false means the normal speed should be ToggleSpeed? If IsMPGameSpeedUpAllowed is false we shouldn't be changing speed anyway, but even so the normal speed for MP will be GetDefaultMPGameSpeed?